guppylang/checker/expr_checker.py

```diff
@@ -944,7 +945,7 @@ def check_inout_arg_place(place: Place, ctx: Context, node: PlaceNode) -> Place:
                 exp_sig,
                 True,
             )
-            return replace(place, setitem_call=setitem_call)
+            return replace(place, setitem_call=(setitem_call, rhs))
```

Comment:
RHS is not a variable. We should create a new temporary to satisfy the contract with `setitem_call`